src/main/java/lrgs/multistat/AlarmList.java

```diff
@@ -0,0 +1,152 @@
+package lrgs.multistat;
+
+import java.util.*;
+
+import javax.swing.*;
+
+public class AlarmList
+	extends SpinnerListModel
+{
+	Vector alarmList;
+	CancelledAlarmList cancelledAlarmList;
+
+	public AlarmList()
+	{
+		super();
+		alarmList = new Vector();
+		/*
+		The SpinnerListModel requires a non-empty list. Therefore we add one
+		'empty' alarm at the beginning of the list which is always there.
+		*/
+		alarmList.add(new Alarm("", "", "", "", 0, "", true));
+		setList(alarmList);
+		cancelledAlarmList = new CancelledAlarmList();
+	}
+
+	/**
+	 * Adds an alarm to the list. If an alarm with this module and event num
+	 * is already in the list, it is replaced with the passed alarm.
+	 * @param alarm the alarm.
+	 */
+	public synchronized void addAlarm(final Alarm alarm)
+	{
+		/* Don't add, if this alarm is already in the list, just replace. */
+		for(int idx = 0; idx < alarmList.size(); idx++)
+		{
+			Alarm a = (Alarm)alarmList.get(idx);
+			if (a.module.equals(alarm.module) && a.alarmNum == alarm.alarmNum)
+			{
+				alarm.assertionCount = a.assertionCount + 1;
+				alarmList.setElementAt(alarm, idx);
+				notifyGui();
+				return;
+			}
+		}
+
+		/* Add the alarm. */
+		alarmList.add(alarm);
+		if (getNumAlarms() == 1)
+// Already in GUI thread.
+//			SwingUtilities.invokeLater(
+//				new Runnable()
+//			{
+//				public void run()
+//				{
+					setValue(alarm);
```

Comment:
This is rather nested for a straight function call. combined with the already in GUI thread comment it's probably reasonable to just deleted the commented code.

Reply:
Done.